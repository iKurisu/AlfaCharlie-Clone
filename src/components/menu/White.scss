.menu-white {
  position: absolute;
  width: 100%;
  height: 100%;
  background: white;
}
