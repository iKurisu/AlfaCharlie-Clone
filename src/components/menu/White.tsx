import React, { useRef } from "react";
import useTransition from "hooks/useTransition";
import useDidUpdateEffect from "hooks/useDidUpdateEffect";
import "./White.scss";

interface Props {
  isOpen: boolean;
}

const White = ({ isOpen }: Props): JSX.Element => {
  const element = useRef<HTMLDivElement>(null);

  const fadeIn = useTransition(element, {
    from: { transform: `translateX(100%)`, opacity: 1 },
    to: { transform: `translateX(0)` },
    config: {
      duration: 400,
      timing: [0.25, 0.1, 0.25, 1]
    }
  });

  const reset = useTransition(element, {
    from: { transform: "translateX(0)" },
    to: { transform: `translateX(100%)` },
    config: { duration: 0, delay: 850 }
  });

  useDidUpdateEffect((): void => {
    if (isOpen) {
      fadeIn();
    } else {
      reset();
    }
  }, [isOpen]);

  return (
    <div
      className="menu-white"
      style={{ transform: `translateX(100%)`, opacity: 1 }}
      ref={element}
    />
  );
};

export default White;
