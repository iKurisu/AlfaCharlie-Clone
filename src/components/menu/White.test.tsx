import React from "react";
import renderer from "react-test-renderer";
import White from "./White";

it("renders correctly", (): void => {
  const background = renderer.create(<White isOpen={true} />);

  expect(background).toMatchSnapshot();
});
