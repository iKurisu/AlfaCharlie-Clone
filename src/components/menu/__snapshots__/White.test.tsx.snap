// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="menu-white"
  style={
    Object {
      "opacity": 1,
      "transform": "translateX(100%)",
    }
  }
/>
`;
